llvm/lib/Transforms/Scalar/LoopInterchange.cpp

```diff
@@ -416,11 +424,11 @@ struct LoopInterchange {
 
   bool processLoopList(SmallVectorImpl<Loop *> &LoopList) {
     bool Changed = false;
-    unsigned LoopNestDepth = LoopList.size();
-    if (LoopNestDepth < 2) {
-      LLVM_DEBUG(dbgs() << "Loop doesn't contain minimum nesting level.\n");
+
+    if (!hasMinimumLoopDepth(LoopList))
```

Comment:
If `LoopNestDepth` is smaller than 2 then we should already have returned from `if (!hasMinimumLoopDepth(LoopList))` on line 1727, which indicates the check on line 428 can be deleted?

Reply:
Yes, that is correct. I wanted the function to do its check before proceeding. I can probably convert it to an assert or remove it altogether. What do you think?

Reply:
Yes I think it makes sense to convert it to an assert here.

Reply:
Done